pkg/config/service-account_test.go

```diff
@@ -65,15 +61,7 @@ var _ = Describe("ServiceAccount", func() {
 	})
 
 	It("Should create a new service account iso disk without a Disk device", func() {
-		vmi := api.NewMinimalVMI("fake-vmi")
-		vmi.Spec.Volumes = append(vmi.Spec.Volumes, v1.Volume{
-			Name: "serviceaccount-volume",
-			VolumeSource: v1.VolumeSource{
-				ServiceAccount: &v1.ServiceAccountVolumeSource{
-					ServiceAccountName: "testaccount",
-				},
-			},
-		})
+		vmi := libvmi.New()
```

Comment:
You are missing `VolumeSource` here

Reply:
like use this
```go
vmi.Spec.Volumes = append(vmi.Spec.Volumes, v1.Volume{
			Name: "serviceaccount-volume",
			VolumeSource: v1.VolumeSource{
				ServiceAccount: &v1.ServiceAccountVolumeSource{
					ServiceAccountName: "testaccount",
				},
			},
		})
```
or should i make option ??

Reply:
No option required here, it is a very specific case

Reply:
got it